airrohr-firmware/html-content.h

```diff
@@ -27,6 +27,7 @@ const char DBG_TXT_SDS011_VERSION_DATE[] PROGMEM = "SDS011 version date";
 const char DBG_TXT_CONNECTING_TO[] PROGMEM = "Connecting to ";
 const char DBG_TXT_FOUND[] PROGMEM = " ... found";
 const char DBG_TXT_NOT_FOUND[] PROGMEM = " ... not found";
+const char DBG_TXT_HR[] PROGMEM = "-----";
```

Comment:
Replacing strings less or equal 4 bytes doesn't save space but needs a functions call.
In some cases you even doesn't save any space for longer strings, or you need more memory.

Reply:
the assembly code *length* is exactly identical between:

```c++
debug_out("foo");
```

and 

```c++
debug_out(FPSTR("foo"));
```

thats because the former is actually implicitly doing this:

```c++
debug_out(String("foo"));
```

so we're just treating one function call over the other. in the future I'd like to add specialisations to debug_* to avoid the implicit String() construction overhead (which is both code size and runtime suboptimal). so I'll drop this again.

Reply:
Thanks for clarification.